packages/visx-xychart/src/components/series/private/BaseBarStack.tsx

```diff
@@ -0,0 +1,224 @@
+
+      // invoke showTooltip for each key so all data is available in context,
+      // and let Tooltip find the nearest point among them
+      dataKeys.forEach(key => {
+        const entry = dataRegistry.get(key);
+        const childData = barSeriesChildren.find(child => child.props.dataKey === key)?.props.data;
+        if (childData && svgPoint && width && height && showTooltip) {
+          const datum = findNearestStackDatum(
+            {
+              point: svgPoint,
+              data: entry.data,
+              xScale,
+              yScale,
+              xAccessor: entry.xAccessor,
+              yAccessor: entry.yAccessor,
+              width,
+              height,
+            },
+            childData,
+            horizontal,
+          );
+
+          if (datum) {
+            showTooltip({
+              key,
+              svgPoint,
+              ...datum,
+            });
+          }
+        }
+      });
+    },
+    [
+      barSeriesChildren,
+      dataRegistry,
+      dataKeys,
+      horizontal,
+      xScale,
+      yScale,
+      width,
+      height,
+      showTooltip,
+    ],
+  );
+  useEventEmitter('mousemove', handleMouseMove);
+  useEventEmitter('mouseout', hideTooltip);
+
+  // if scales and data are not available in the registry, bail
+  if (dataKeys.some(key => dataRegistry.get(key) == null) || !xScale || !yScale || !colorScale) {
+    return null;
+  }
+
+  const barThickness = getBandwidth(horizontal ? yScale : xScale);
+
+  const bars = stackedData
+    .flatMap((barStack, stackIndex) => {
+      const entry = dataRegistry.get(barStack.key);
+      if (!entry) return null;
+
+      return barStack.map((bar, index) => {
+        const barLength = horizontal
+          ? (xScale(getSecondItem(bar)) || 0) - (xScale(getFirstItem(bar)) || 0)
+          : (yScale(getFirstItem(bar)) || 0) - (yScale(getSecondItem(bar)) || 0);
+
+        const barY = horizontal
+          ? 'bandwidth' in yScale
+            ? yScale(getStackValue(bar.data))
+            : Math.max((yScale(getStackValue(bar.data)) || 0) - barThickness / 2)
+          : yScale(entry.yAccessor(bar));
+
+        const barX: number | undefined = horizontal
+          ? xScale(getFirstItem(bar))
+          : 'bandwidth' in xScale
+          ? xScale(getStackValue(bar.data))
+          : Math.max((xScale(getStackValue(bar.data)) || 0) - barThickness / 2);
+
+        return isValidNumber(barX) && isValidNumber(barY)
+          ? {
+              key: `${stackIndex}-${barStack.key}-${index}`,
+              x: barX ?? 0,
+              y: barY ?? 0,
+              width: horizontal ? barLength : barThickness,
+              height: horizontal ? barThickness : barLength,
+              fill: colorScale(barStack.key),
+            }
+          : null;
+      });
+    })
+    .filter(bar => bar) as Bar[];
+
+  return (
+    <g className="visx-bar-stack">
+      <BarsComponent bars={bars} horizontal={horizontal} xScale={xScale} yScale={yScale} />
```

Comment:
this is all the same as previous `BarStack`, `bars` are just passed to `BarsComponent` now.